src/app-logic/url-handling.js

```diff
@@ -593,6 +593,10 @@ function parseLocalTrackOrder(rawText: string): Map<Pid, TrackIndex[]> {
   return localTrackOrderByPid;
 }
 
+class UrlUpgradeError extends Error {
```

Comment:
It would be good to add a small comment explains that this is used to identify and show the error to the screen if upgrade fails.